src/ir/from_ast/astconv.rs

```diff
@@ -163,22 +163,18 @@ impl<'ctx, 'prog> BuildCtx<'ctx, 'prog> {
                 self.comp.add(e)
             }
             ast::Expr::Op { op, left, right } => {
-                let l = self.expr(*left);
-                let r = self.expr(*right);
-                match op {
-                    ast::Op::Add => l.add(r, self.comp),
-                    ast::Op::Mul => l.mul(r, self.comp),
-                    ast::Op::Sub => l.sub(r, self.comp),
-                    ast::Op::Div => l.div(r, self.comp),
-                    ast::Op::Mod => l.rem(r, self.comp),
-                }
+                let lhs = self.expr(*left);
+                let rhs = self.expr(*right);
+                // The .add call simplifies the expression if possible
+                self.comp.add(ir::Expr::Bin { op, lhs, rhs })
```

Comment:
This needs to be changed back because it might cause a stackoverflow right?

Reply:
Not as far as I can tell? `add()` doesn't simplify things, I just made separate functions that will simplify an expression

Reply:
This is handled by #237